ut/columns_ut.cpp

```diff
@@ -114,6 +152,135 @@ TEST(ColumnsCase, DateAppend) {
     ASSERT_EQ(col2->At(0), (now / 86400) * 86400);
 }
 
+TEST(ColumnsCase, DateTime64_0) {
+    auto column = std::make_shared<ColumnDateTime64>(0ul);
+
+    ASSERT_EQ(Type::DateTime64, column->Type()->GetCode());
+    ASSERT_EQ("DateTime64(0)", column->Type()->GetName());
+    ASSERT_EQ(0u, column->GetPrecision());
+    ASSERT_EQ(0u, column->Size());
+}
+
+TEST(ColumnsCase, DateTime64_6) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    ASSERT_EQ(Type::DateTime64, column->Type()->GetCode());
+    ASSERT_EQ("DateTime64(6)", column->Type()->GetName());
+    ASSERT_EQ(6u, column->GetPrecision());
+    ASSERT_EQ(0u, column->Size());
+}
+
+TEST(ColumnsCase, DateTime64_Append_At) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    const auto data = MakeDateTime64s();
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    ASSERT_EQ(data.size(), column->Size());
+    for (size_t i = 0; i < data.size(); ++i) {
+        ASSERT_EQ(data[i], column->At(i));
+    }
+}
+
+TEST(ColumnsCase, DateTime64_Clear) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    // Clearing empty column doesn't crash and produces expected result
+    ASSERT_NO_THROW(column->Clear());
+    ASSERT_EQ(0u, column->Size());
+
+    const auto data = MakeDateTime64s();
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    ASSERT_NO_THROW(column->Clear());
+    ASSERT_EQ(0u, column->Size());
+}
+
+TEST(ColumnsCase, DateTime64_Swap) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    const auto data = MakeDateTime64s();
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    auto column2 = std::make_shared<ColumnDateTime64>(6ul);
+    column->Swap(*column2);
+
+    // Validate that all items were transferred to column2.
+    ASSERT_EQ(0u, column->Size());
+    ASSERT_EQ(data.size(), column2->Size());
+
+    // Clearing doesn't affects precision value.
+    ASSERT_EQ(6ul, column->GetPrecision());
+
+    for (size_t i = 0; i < data.size(); ++i) {
+        ASSERT_EQ(data[i], column2->At(i));
+    }
+}
```

Comment:
This checks only that one direction is OK after the swap, right? One typical bug of a `swap` implementation, is when after a swap the destinations contains what was in the source, but the source is left empty.

Reply:
fixed